src/definitions/bluebps.json

```diff
@@ -0,0 +1,254 @@
+            },
+            "current_max": {
+                "id": "9002",
+                "description": "Set the maximum allowed battery current",
+                "payload": [
+                    {
+                        "name": "limit",
+                        "type": "u16",
+                        "description": "The maximum allowed battery current 0~20000 = 0~200A",
+                        "units": "centiamp"
+                    }
+                ]
+            },
+            "current_timeout": {
+                "id": "9003",
+                "description": "Set the over-current timeout",
+                "payload": [
+                    {
+                        "name": "timeout",
+                        "type": "u16",
+                        "description": "If the battery current exceeds the configured limit for this duration of time, the power will be locked-out",
+                        "units": "millisecond"
+                    }
+                ]
+            },
+            "temperature_max": {
+                "id": "9004",
+                "description": "Set the maximum allowed battery temperature",
+                "payload": [
+                    {
+                        "name": "limit",
+                        "type": "u16",
+                        "description": "The minimum voltage allowed for any individual cell. 0~5000: 0~5V",
+                        "units": "centidegrees Centigrade"
+                    }
+                ]
+            },
+            "temperature_timeout": {
+                "id": "9005",
+                "description": "Set the over-temperature timeout",
+                "payload": [
+                    {
+                        "name": "timeout",
+                        "type": "u16",
+                        "description": "If the battery temperature exceeds the configured limit for this duration of time, the power will be locked-out",
+                        "units": "millisecond"
+                    }
+                ]
+            },
+            "stream_rate": {
+                "id": "9006",
+                "description": "Set the frequency to automatically output state messages.",
+                "payload": [
+                    {
+                        "name": "rate",
+                        "type": "u16",
+                        "description": "Rate to stream `state` messages. 0~1000Hz",
+                        "units": "Hz"
+                    }
+                ]
+            },
+            "lpf_setting": {
+                "id": "9007",
+                "description": "Low pass filter setting. 0.0 = no filtering, 0.99 = heavy filtering",
+                "payload": [
+                    {
+                        "name": "setting",
+                        "type": "float",
+                        "description": "0.0~0.99"
+                    }
+                ]
+            }
+        },
+        "get": {
+            "cell_voltage_min": {
```

Comment:
Maybe a single message the return all sensor states ? E.g: Limits and timeouts.
Probably this will cover 90% of our and user case.

Reply:
I did draft it this way also, and I favored the individual messages for the individual settings.

The reason is that the individual settings are largely independent of each other, unlike ping360 where transmit duration, transmit frequency and etc must all be selected simultaneously for an atomic process.

BPS individual settings are allowed to be set individually, one by one, while the device continues to sample and operate normally between each message, so I allow that.

There are advantages/disadvantages to both ways, I'm still open to the combined single message for all limits+timeout, but I have been thinking about it already :)

Reply:
I'm talking about a single get message, set is not necessary as you said.

Reply:
aha! that's a good idea.

Reply:
Part of the reason I did it as (matched) individual commands is this was making the template implementations for device set/get easier to match, every function returns a single value this way. I think we should try the api that gets generated next and see if it can be improved.